### PR TITLE
Fix java_release workflow by removing step without users/with

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:
This has been broken for a few commits now: https://github.com/feast-dev/feast/actions/workflows/java_release.yml

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: java/serving/src/main/java/feast/serving/config/RegistryConfig.java [hunk 1]
```diff
@@ -59,7 +58,7 @@ RegistryFile registryFile(FeastProperties feastProperties, ApplicationContext co
         return new S3RegistryFile(context.getBean(AmazonS3.class), registryPath);
       case "":
       case "file":
-        return new LocalRegistryFile(Paths.get(registryPath));
+        return new LocalRegistryFile(registryPath);
```
Comment (woop) [inline java/serving/src/main/java/feast/serving/config/RegistryConfig.java:61]: Can you explain this change?

Comment (achals) [reply java/serving/src/main/java/feast/serving/config/RegistryConfig.java:61]: Trying to address this error in java unit tests:

```
Error:  /home/runner/work/feast/feast/java/serving/src/main/java/feast/serving/config/RegistryConfig.java:[62,47] incompatible types: java.nio.file.Path cannot be converted to java.lang.String
```

Comment (pyalex) [reply java/serving/src/main/java/feast/serving/config/RegistryConfig.java:61]: Yeah, this is bug from my PR #2043, have no idea how it passed tests

File: .github/workflows/java_release.yml [hunk 2]
```diff
@@ -77,22 +77,29 @@ jobs:
...
+        run: make build-${{ matrix.component }}-docker REGISTRY=${REGISTRY} VERSION=${RELEASE_VERSION}
         env:
           RELEASE_VERSION: ${{ needs.get-version.outputs.release_version }}
           VERSION_WITHOUT_PREFIX: ${{ needs.get-version.outputs.version_without_prefix }}
           HIGHEST_SEMVER_TAG: ${{ needs.get-version.outputs.highest_semver_tag }}
-      - name: Build image
-        run: make build-${{ matrix.component }}-docker REGISTRY=${REGISTRY} VERSION=${RELEASE_VERSION}
       - name: Push image
         run: make push-${{ matrix.component }}-docker REGISTRY=${REGISTRY} VERSION=${RELEASE_VERSION}
+        env:
```
Comment (pyalex) [inline .github/workflows/java_release.yml:88]: what the point to duplicate env here?

Comment (achals) [reply .github/workflows/java_release.yml:88]: The bare `env` version step up above is breaking the workflow: https://github.com/feast-dev/feast/actions/workflows/java_release.yml

**Pull request merged**